Fix xnnpack pybind

mcr229 opened the pull request:
Fixing XNNPACK Pybind.

Currently pybindings for xnnpack fail due to uninitialized error. We notice the following:
```
(gdb) info symbol xnn_initialize
xnn_initialize in section .text of /home/maxren/.conda/envs/executorch/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so
(gdb) info symbol xnn_create_subgraph
xnn_create_subgraph in section .text of /home/maxren/.conda/envs/executorch/lib/python3.10/site-packages/executorch/extension/pybindings/portable_lib.cpython-310-x86_64-linux-gnu.so
```

That is xnn_initialize symbole comes from libtorch_cpu, but when we call xnn_create_subgraph we call it from another library. It seems like explicitly adding XNNPACK to the portable_libs pybind fixes this problem.

Additionaly, we fix some fPIC issues when building XNNPACK.

mcr229 committed:
[XNNPACK] Set PIC Flag for CMake

mcr229 committed:
Fix XNNPACK Pybind

metascroy reviewed (commented):
LGTM...you tested this fixes the pybind xnnpack issue?

larryliu0820 committed:
[pybind] Add pybind w/ xnnpack and run unit tests in CI

Summary: As titled. Add executorch/backends/xnnpack/test/models
directory into pytest.ini

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
